pkg/controllers/securitypolicy_controller.go

```diff
@@ -68,6 +73,74 @@ func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return ctrl.Result{}, nil
 }
 
+func (r *SecurityPolicyReconciler) setSecurityPolicyStatusCreated(ctx *context.Context, sec_policy *v1alpha1.SecurityPolicy) {
+	newConditions := []v1alpha1.SecurityPolicyCondition{
+		{
+			Type:    v1alpha1.SecurityPolicyCreated,
+			Status:  v1.ConditionTrue,
+			Message: "NSX Security Policy has been successfully created/updated",
+			Reason:  "NSX API returned 200 response code for PATCH",
+		},
+		{
+			Type:   v1alpha1.SecurityPolicyError,
+			Status: v1.ConditionFalse,
+		},
+	}
+	r.updateSecurityPolicyStatusConditions(ctx, sec_policy, newConditions)
+}
+
+func (r *SecurityPolicyReconciler) setSecurityPolicyStatusError(ctx *context.Context, sec_policy *v1alpha1.SecurityPolicy, err *error) {
+	newConditions := []v1alpha1.SecurityPolicyCondition{
+		{
+			Type:    v1alpha1.SecurityPolicyError,
+			Status:  v1.ConditionTrue,
+			Message: "NSX Security Policy could not be created/updated",
+			Reason:  fmt.Sprintf("Error occurred while processing the Security Policy CRD. Please check the config and try again. Error: %v", *err),
+		},
+	}
+	r.updateSecurityPolicyStatusConditions(ctx, sec_policy, newConditions)
+}
+
+func (r *SecurityPolicyReconciler) updateSecurityPolicyStatusConditions(ctx *context.Context, sec_policy *v1alpha1.SecurityPolicy, newConditions []v1alpha1.SecurityPolicyCondition) {
+	conditionsUpdated := false
+	for i := range newConditions {
+		if r.mergeSecurityPolicyStatusCondition(ctx, sec_policy, &newConditions[i]) {
+			conditionsUpdated = true
+		}
+	}
+	if conditionsUpdated {
+		r.Client.Status().Update(*ctx, sec_policy)
+		log.V(1).Info(fmt.Sprintf("Updated Security Policy CRD with name \"%s\" in ns \"%s\" status Conditions", sec_policy.Name, sec_policy.Namespace), "New Conditions", newConditions)
+	}
+}
+
+func (r *SecurityPolicyReconciler) mergeSecurityPolicyStatusCondition(ctx *context.Context, sec_policy *v1alpha1.SecurityPolicy, newCondition *v1alpha1.SecurityPolicyCondition) bool {
+	matchedCondition := getExistingConditionOfType(newCondition.Type, sec_policy.Status.Conditions)
+
+	if reflect.DeepEqual(matchedCondition, newCondition) {
+		log.V(1).Info("Conditions already match", "New Condition", newCondition, "Existing Condition", matchedCondition)
+		return false
+	}
+
+	if matchedCondition != nil {
+		matchedCondition.Reason = newCondition.Reason
+		matchedCondition.Message = newCondition.Message
+		matchedCondition.Status = newCondition.Status
+	} else {
+		sec_policy.Status.Conditions = append(sec_policy.Status.Conditions, *newCondition)
```

Comment:
I don't think we can just merge two conditions. Like if the existing condition is `Created`, then we want to update an `Error` condition. I think it's confusing to user as there are both good and bad conditions.
In other CRD status, we have only one type `Ready`, status is `True` or `False` based on the realization succeeded or failed.

Reply:
ok, let me update it to just use `Ready`